project/externals/boost.cmake

```diff
@@ -5,13 +5,14 @@
 
 set(name boost)
 set(source_dir ${CMAKE_CURRENT_BINARY_DIR}/${name}/source)
+string(SUBSTRING "${CMAKE_HOST_SYSTEM_PROCESSOR}" 0 4 processor_prefix)
```

Comment:
I don't think you need to do it here again. The code here should be able to see the one defined in the upper level CMakeList.txt